### ISSUE TITLE
Catch "File size exceeds maximum allowed size"

### ISSUE THREAD

**Issue by perifer** [mjackson/multipart-parser#8]
I expected this error to be caught with a try/catch:

`MultipartParseError: File size exceeds maximum allowed size of 10485760 bytes`

Tested with https://github.com/mjackson/multipart-parser/blob/main/examples/node/server.js. When the error occurs the server crashes.

**Comment by mjackson** [comment]
Everything works fine for me on node 20.15.0. Which version of node are you on?

I did just push [an update to the node example](40b446068e578d9e4e1dd360e8412eb71eb0f566) that adds the `Connection: close` header to the response in case of an error, so the input stream doesn't hang. But I don't think that's related.

**Comment by perifer** [comment]
Thanks for the feedback!

I tested it on Node 22.1.0. I just tried with 20.16.0 with the same result. Another error "MultipartParseError: Request is not a multipart request", is caught correctly when I trigger an error by removing `enctype="multipart/form-data"` from the form in the example.

Here is the terminal output with the error when I run `pnpm start` and then uploads a big file in Safari or Chrome, if that helps.

```
pnpm start

> multipart-parser-node-example@ start /Users/perifer/Downloads/multipart-parser-main/examples/node
> node server.js

Server listening on http://localhost:3000 ...
file:///Users/perifer/Downloads/multipart-parser-main/dist/lib/multipart.js:302
                throw new MultipartParseError(`File size exceeds maximum allowed size of ${this.#maxFileSize} bytes`);
                      ^

MultipartParseError: File size exceeds maximum allowed size of 10485760 bytes
    at #writeBody (file:///Users/perifer/Downloads/multipart-parser-main/dist/lib/multipart.js:302:23)
    at MultipartParser.push (file:///Users/perifer/Downloads/multipart-parser-main/dist/lib/multipart.js:241:36)
    at MultipartParser.parse (file:///Users/perifer/Downloads/multipart-parser-main/dist/lib/multipart.js:134:43)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v20.16.0
ELIFECYCLE  Command failed with exit code 1.
```

**Comment by mjackson** [comment]
I'm sorry but I just tried running the node example on node 22.1.0 and 20.16.0 and I can't reproduce the issue you're experiencing on either one.

FWIW I just released version 0.4.0 which may give you slightly different results if you'd care to check again.

I'm going to sync with my team tomorrow and ask if anyone else can reproduce what you're experiencing.

**Comment by perifer** [comment]
I did some more test. I could trigger the error by changing the maxFileSize setting. With a low enough setting, the server never crashed. But as I increased it the server started crashing. Here is a codesandbox that I could reproduce it on:

https://codesandbox.io/p/devbox/stoic-sea-72cmwm

maxFileSize is set to 100000. With that high setting it seems to crash consistently. Less so with a lower setting. At least it seems so in my limited testing.

I tried with this image: https://en.wikipedia.org/wiki/Cat#/media/File:Cat_August_2010-4.jpg
